frontend/test/pytest/test_device_api.py

```diff
@@ -100,6 +124,24 @@ def test_qjit_device():
     assert len(device_qjit.observables) > 0
 
 
+@pytest.mark.skipif(
+    not pathlib.Path(get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/libdummy_device.so").is_file(),
+    reason="lib_dummydevice.so was not found.",
+)
```

Comment:
It looks like this test could run even without the dummy device being present, since the check happens right on initialization (before `config` and `backend_info` are used).

Reply:
We could also think about always building the dummy device with the runtime, that way we always have at least one device available for tests like this. What do you think?

Reply:
Yes let me do that in another PR, good idea :+1:

Reply:
Oh I remember why it is like that, it is because we don't build it with wheels. So I think it works well because it was added for wheels testing. Do you want us to build it in the wheels and remove the skips?

Reply:
So I think we could totally have the dummy device be always present (in dev builds and wheels), but also for this particular test case I don't think you need to skip because the error we are testing for is raised before Catalyst realizes the device is not there (e.g. if `config` and `backend_info` were just `None`).

Reply:
We have ```config = device_get_toml_config(device)
    backend_info = extract_backend_info(device, config)``` in the test and cannot pass without the shared library.

Reply:
Yes but my point is those values are not used and could just set them to `None`.